pystac_monty/sources/ifrc_events.py

```diff
@@ -0,0 +1,163 @@
+        if not ifrc_data:
+            return []
+
+        for data in ifrc_data:
+            item = self.make_source_event_item(data=data)
+            items.append(item)
+        return items
+
+    def make_source_event_item(self, data: dict) -> Item:
+        """Create ane event item"""
+        geometry = None
+        bbox = None
+        geom_data = self.geocoder.get_geometry_by_country_name(data["country"]["iso3"])
+
+        if geom_data:
+            geometry = geom_data["geometry"]
+            bbox = geom_data["bbox"]
+
+        # Create item
+        item = Item(
+            id=f"{STAC_EVENT_ID_PREFIX}{data["aid"]}",
+            geometry=geometry,
+            bbox=bbox,
+            datetime=data["start_date"],
+            start_datetime=data["start_date"],
+            end_datetime=data["end_date"],
+            properties={
+                    "title": data[""],
+            },
+        )
+
+        # Add Monty extension
+        MontyExtension.add_to(item)
+        monty = MontyExtension.ext(item)
+        monty.episode_number = 1  # IFRC DREF doesn't have episodes
+        monty.hazard_codes = self._map_ifrc_to_hazard_codes(data["dtype"]["name"])
+        monty.country_codes = data["country"]["iso3"]
+        monty.compute_and_set_correlation_id(hazard_profiles=self.hazard_profiles)
+
+        # Set collection and roles
+        item.set_collection(self.get_event_collection())
+        item.properties["roles"] = ["source", "event"]
+
+        # Filter out relevant disaster types
+        monty_accepted_disaster_types = {
+            "Earthquake", "Cyclone", "Volcanic Eruption", "Tsunami", "Flood",
+            "Cold Wave", "Fire", "Heat Wave", "Drought", "Storm Surge",
+            "Landslide", "Flash Flood"
+        }
+        if data["dtype"]["name"] not in monty_accepted_disaster_types:
+            return []
+
+        return item
+
+    def _map_ifrc_to_hazard_codes(self, classification_key: str) -> List[str]:
+        """
+        Map EM-DAT classification key to UNDRR-ISC 2020 Hazard codes
+
+        Args:
+            classification_key: dtype name (e.g., 'Flood')
+
+        Returns:
+            List of UNDRR-ISC hazard codes
+        """
+        # IFRC DREF hazards classification mapping to UNDRR-ISC codes
+        mapping = {
+            "Earthquake": ["GH0001", "GH0002", "GH0003", "GH0004", "GH0005"],
+            "Cyclone": ["MH0030", "MH0031", "MH0032"],
+            "Volcanic Eruption": ["GH009", "GH0013", "GH0014", "GH0015", "GH0016"],
+            "Tsunami": ["MH0029", "GH0006"],
+            "Flood": ["FL"],  # General flood
+            "Cold Wave": "MH0040",
+            "Fire": ["FR"],
+            "Heat Wave": ["MH0047"],
+            "Drought": ["MH0035"],
+            "Storm Surge": ["MH0027"],
+            "Landslide": ["GH0007"],
+            "Flash Flood": ["MH0006"],
+            "Epidemic": ["EP"],  # General epidemic
+        }
+
+        if not classification_key:
+            return []
```

Comment:
lets do this in the begining so that when classification_key is none no more calculation takes place.